src/SmartComponents/Modals/UnassignSystemsModal.test.js

```diff
@@ -62,35 +54,25 @@ describe('UnassignSystemsModal', () => {
             })
         );
 
-        await wrapper.update().find(Modal).props().actions[0].props.onClick();
-        expect(addNotification).toHaveBeenCalledWith(
-            patchSetUnassignSystemsNotifications(1).success
-        );
-        expect(unassignSystemsModalState).toEqual({ isUnassignSystemsModalOpen: false, shouldRefresh: true, systemsIDs: [] });
-        expect(unassignSystemFromPatchSet).toHaveBeenCalledWith({ inventory_ids: ['test_1'] });
-    });
+        fireEvent.click(screen.getByText('Remove'));
 
-    it('should close the modal', () => {
-        unassignSystemsModalState.isUnassignSystemsModalOpen = true;
-        act(() => {
-            wrapper.props().setUnassignSystemsModalOpen(false);
+        await waitFor(() => {
+            expect(addNotification).toHaveBeenCalledWith(
+                patchSetUnassignSystemsNotifications(1).success
+            );
+            expect(unassignSystemsModalState).toEqual({ isUnassignSystemsModalOpen: false, shouldRefresh: true, systemsIDs: [] });
+            expect(unassignSystemFromPatchSet).toHaveBeenCalledWith({ inventory_ids: ['test_1'] });
         });
-
-        wrapper.update();
-        expect(unassignSystemsModalState.isUnassignSystemsModalOpen).toBeFalsy();
     });
 
-    it('should hide the modal when isUnassignSystemsModalOpen=false', () => {
-        const wrapper = mountWithIntl(<UnassignSystemsModal
-            unassignSystemsModalState={{
-                isUnassignSystemsModalOpen: false,
-                systemsIDs: ['test_1', 'test_2', 'test_3']
-            }}
-            setUnassignSystemsModalOpen={setUnassignSystemsModalOpen}
-        />
-        );
+    it('should close the modal', async () => {
+        unassignSystemsModalState.isUnassignSystemsModalOpen = true;
+        screen.debug(undefined, 30000);
+        fireEvent.click(screen.getByLabelText('Close'));
```

Comment:
nit-pick:
```suggestion
        userEvent.click(screen.getByLabelText('Close'));
```

Reply:
I do not see a huge gain in using `userEvent` when I need to only simulate the onClick event.

Reply:
> user-event allows you to describe a user interaction instead of a concrete event. It adds visibility and interactability checks along the way and manipulates the DOM just like a user interaction in the browser would. It factors in that the browser e.g. wouldn't let a user click a hidden element or type in a disabled text box.
> This is [why you should use user-event](https://ph-fritsche.github.io/blog/post/why-userevent) to test interaction with your components.